web/modules/custom/collection/src/Plugin/Block/CollectionContentBlock.php

```diff
@@ -87,7 +87,9 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function build() {
     if (empty($this->collection)) {
-      throw new \Exception('The "Collection content" block can only be shown on collection pages.');
+      // If this is not a collection page, return an empty array so that the
+      // block is not rendered.
+      return [];
```

Comment:
The idea behind this was to not render the block if the collection is not retrieved from the route.
I have put back the exception and moved the check to the blockAccess method.